services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py

```diff
@@ -0,0 +1,312 @@
+import logging
+from collections import deque
+from typing import Any, Awaitable, Final, Optional
+
+from fastapi import FastAPI
+from pydantic import BaseModel
+from servicelib.fastapi.long_running_tasks.server import TaskProgress
+from servicelib.utils import logged_gather
+from simcore_sdk.node_data import data_manager
+
+from ..core.docker_compose_utils import (
+    docker_compose_down,
+    docker_compose_pull,
+    docker_compose_restart,
+    docker_compose_rm,
+    docker_compose_up,
+)
+from ..core.docker_logs import start_log_fetching, stop_log_fetching
+from ..core.rabbitmq import RabbitMQ
+from ..core.settings import ApplicationSettings
+from ..core.utils import assemble_container_names
+from ..core.validation import validate_compose_spec
+from ..models.schemas.application_health import ApplicationHealth
+from ..models.shared_store import SharedStore
+from ..modules import nodeports
+from ..modules.directory_watcher import directory_watcher_disabled
+from ..modules.mounted_fs import MountedVolumes
+
+logger = logging.getLogger(__name__)
+
+
+class ContainersCreate(BaseModel):
+    docker_compose_yaml: str
+
+
+async def send_message(rabbitmq: RabbitMQ, message: str) -> None:
+    logger.info(message)
+    await rabbitmq.post_log_message(f"[sidecar] {message}")
+
+
+# TASKS
+
+# NOTE: most services have only 1 "working" directory
+CONCURRENCY_STATE_SAVE_RESTORE: Final[int] = 2
+
+
+async def task_create_service_containers(
+    progress: TaskProgress,
+    settings: ApplicationSettings,
+    containers_create: ContainersCreate,
+    shared_store: SharedStore,
+    mounted_volumes: MountedVolumes,
+    app: FastAPI,
+    application_health: ApplicationHealth,
+    rabbitmq: RabbitMQ,
+    long_running_compose_timeout: int,
+    validation_timeout: int,
+) -> list[str]:
+    progress.publish(message="validating service spec", percent=0)
+
+    shared_store.compose_spec = await validate_compose_spec(
+        settings=settings,
+        compose_file_content=containers_create.docker_compose_yaml,
+        mounted_volumes=mounted_volumes,
+        docker_compose_config_timeout=validation_timeout,
+    )
+    shared_store.container_names = assemble_container_names(shared_store.compose_spec)
+
+    logger.debug("Validated compose-spec:\n%s", f"{shared_store.compose_spec}")
+
+    await send_message(rabbitmq, "starting service containers")
+    assert shared_store.compose_spec  # nosec
+
+    with directory_watcher_disabled(app):
+        # removes previous pending containers
+        progress.publish(message="cleanup previous used resources")
+        await docker_compose_rm(shared_store.compose_spec, settings)
+
+        progress.publish(message="pulling images", percent=0.01)
+        await docker_compose_pull(
+            shared_store.compose_spec, settings, timeout=long_running_compose_timeout
```

Comment:
how do you define that timeout here??? for pulling? this can take between 0 and X amount of hours depending on the size of the containers. If that timeout is too small it will never work because one image is too big.

Reply:
Agreed with you, the dropped the timeout on the pulling operation.